### PR TITLE
feat(forms): Adds string completion widget

### PR DESCRIPTION
**Pull request by talend-glorieux**
**What is the problem this PR is trying to solve?**
We are in need of a String field with completion from a remote suggestion list. 

**What is the chosen solution to this problem?**
Adds a String completion widget that takes remote suggestion list.

It has the following features:
* Input highlight
* Mouse and keyboard selection
* single or multi sections

Update to the TypeAhead component:
* Maps onFocus property
* Maps Mouse events properties
* Makes multiSection configurable
* Makes autofocus configurable
* Updates component z-index

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/forms/src/widgets/StringCompletionWidget/__snapshots__/StringCompletionWidget.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,343 @@
+exports[`StringCompletionWidget should render 1`] = `
```
Comment (frassinier) [inline packages/forms/src/widgets/StringCompletionWidget/__snapshots__/StringCompletionWidget.test.js.snap]: Does it make sense to check snapshots here?

File: packages/forms/stories/index.js [hunk 2]
```diff
@@ -114,6 +115,43 @@ decoratedStories.add('Multiple actions', () => {
...
+		},
+		uiSchema: {
+			completion: {
+				'ui:widget': 'stringCompletion',
+				'ui:options': {
+					itemsSrc: '/stringCompletionItems.json',
+				},
+			},
+		},
+		properties: {},
```
Comment (frassinier) [inline packages/forms/stories/index.js]: What if I already have properties?

File: packages/forms/package.json [hunk 3]
```diff
@@ -75,9 +75,9 @@
     "lint:es": "eslint --config .eslintrc src",
     "lint:style": "sass-lint -v -q",
     "lint": "npm run lint:es",
-    "storybook": "start-storybook -p 6008",
-    "start": "start-storybook -p 6008",
-    "build-storybook": "build-storybook"
+    "storybook": "start-storybook -s ./stories/static -p 6008",
+    "start": "start-storybook -s ./stories/static -p 6008",
+    "build-storybook": "build-storybook -s ./stories/static"
```
Comment (frassinier) [inline packages/forms/package.json]: Why these 3 lines?

File: packages/forms/stories/static/stringCompletionItems.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+[{
+	"title": "A",
```
Comment (frassinier) [inline packages/forms/stories/static/stringCompletionItems.json]: I'm wondering, this JSON structure is needed by component but I think we can mask complexity for widget?

```json
[
  { "group": "A", "items": [ "Apple", "Apricot"] },
  { "group": "B", "items": [ "Banana"] },
  { "group": "C", "items": [ "Cherry"] }, 
]
```

File: packages/forms/package.json (hunk 3 above)
Comment (talend-glorieux) [inline packages/forms/package.json]: Not required anymore. At the beginning I wanted to load the files with a fetch request.
I'll remove it.

File: packages/forms/src/widgets/StringCompletionWidget/__snapshots__/StringCompletionWidget.test.js.snap (hunk 1 above)
Comment (talend-glorieux) [inline packages/forms/src/widgets/StringCompletionWidget/__snapshots__/StringCompletionWidget.test.js.snap]: Yep I like to unit test the widget.

File: packages/forms/stories/index.js (hunk 2 above)
Comment (talend-glorieux) [inline packages/forms/stories/index.js]: They'll be put in as the value.

File: packages/forms/stories/static/stringCompletionItems.json (hunk 4 above)
Comment (talend-glorieux) [inline packages/forms/stories/static/stringCompletionItems.json]: Yes for sure. This is a first implementation to do quick tests on the integration.
It'll be improved later. The widget API isn't fixed.

**Comment by talend-glorieux** [comment]
I might end up merging the remote items loading part of that widget into the DataListWidget.

**Comment by talend-glorieux** [comment]
Closed in favor of #458

**Pull request closed without merging**